<launch>
    <arg name="id" value="S1" />
    
    <arg name="vehicle" default="iris"/>
    <arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/empty.world"/>
    <!-- gazebo configs -->
    <arg name="gui" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>

    <!-- Gazebo sim -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world_name" value="$(arg world)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
    </include>
    <group ns="$(arg id)">
		<!-- MAVROS and vehicle configs -->
        <arg name="ID" value="0"/>
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn.launch">
            <arg name="x" value="0"/>
            <arg name="y" value="0"/>
            <arg name="z" value="0"/>
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="1.57"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14560"/>
            <arg name="mavlink_tcp_port" value="4560"/>
            <arg name="ID" value="$(arg ID)"/>
            <!-- <arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))"/> -->
            <!-- <arg name="video_uri" value="$(eval 5600 + arg('ID'))"/> -->
            <!-- <arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))"/> -->
        </include>
		<node pkg="px4_path_planner" type="px4_path_planner_node" name="px4_path_planner" output="screen">
			<rosparam command="load" file="$(find px4_path_planner)/params/common.yaml" />
			<param name="wp_file_location" value="$(find px4_path_planner)/path/wp0.csv"/>
			<param name="agent_id" value="$(arg id)"/>
			<param name="nwu_offset_x" value="0.0"/>
            <param name="nwu_offset_y" value="0.0"/>
            <param name="nwu_offset_z" value="0.0"/>
		</node>

		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="udp://:14540@127.0.0.1:14557" />
		</include>

		<node pkg="px4_path_planner" type="px4_path_planner_visualization_node" name="px4_path_planner_visualization" output="screen">
			<param name="agent_id" value="$(arg id)"/>
		</node>
		
		<node type="rviz" name="rviz" pkg="rviz" args="-d $(find px4_path_planner)/rviz/px4_path_planner.rviz" />
    </group>

</launch>
